src/main/scala/com/microsoft/partnercatalyst/fortis/spark/StreamsChangeListener.scala

```diff
@@ -69,50 +69,62 @@ object StreamsChangeListener {
+      System.err.println(s"Service Bus message received ${message}.")
 
       if (message.getEnqueuedTimeUtc.isBefore(this.initializedAt)) {
-        logInfo(s"Service Bus message ignored since it predates listener initialization.")
+        System.err.println(s"Service Bus message ignored since it predates listener initialization.")
         return CompletableFuture.completedFuture(null)
       }
 
       this.scheduledTask match {
         case Some(task) => {
-          logInfo(s"Service Bus message for updated streams received; Re-scheduling streaming context stop for ${settings.sscShutdownDelayMillis} milliseconds from now.")
+          System.err.println(s"Service Bus message for updated streams received; Re-scheduling streaming context stop for ${settings.sscShutdownDelayMillis} milliseconds from now.")
           task.cancel(false)
         }
         case None => {
-          logInfo(s"Service Bus message for updated streams received; Requesting streaming context stop in ${settings.sscShutdownDelayMillis} milliseconds.")
+          System.err.println(s"Service Bus message for updated streams received; Requesting streaming context stop in ${settings.sscShutdownDelayMillis} milliseconds.")
         }
       }
 
       this.currentContext match {
         case Some(context) => {
           this.scheduledTask = Some(this.scheduler.schedule(
-            new ContextStopRunnable(settings, context),
+            new ContextStopRunnable(settings, context, scheduler),
             settings.sscShutdownDelayMillis,
             TimeUnit.MILLISECONDS
           ))
         }
         case None => {
-          logError(s"No streaming context set; Nothing to stop.")
+          System.err.println(s"No streaming context set; Nothing to stop.")
         }
       }
 
       CompletableFuture.completedFuture(null)
     }
   }
 
-  private[spark] class ContextStopRunnable(settings: FortisSettings, ssc: StreamingContext) extends Runnable with Loggable {
+  private[spark] class ContextStopRunnable(settings: FortisSettings, ssc: StreamingContext, scheduler: ExecutorService) extends Runnable with Loggable {
     override def run(): Unit = {
       StreamsChangeListener.suggestedExitCode = 10
-      logInfo(s"Requesting streaming context stop now.")
-      ssc.stop(stopSparkContext = true, stopGracefully = false)
-      logInfo(s"Streaming context stop complete; Cleaning up...")
-      if (!settings.progressDir.isEmpty) {
-        Path(settings.progressDir).deleteRecursively()
+      System.err.println(s"Requesting streaming context stop now.")
+      val timeoutTask = scheduler.submit(new TimeoutRunnable)
+      try {
+        ssc.stop(stopSparkContext = true, stopGracefully = false)
+        timeoutTask.cancel(false)
+        System.err.println(s"Streaming context stop complete; Cleaning up...")
+        if (!settings.progressDir.isEmpty) {
+          Path(settings.progressDir).deleteRecursively()
+        }
```

Comment:
Moving these lines into the try/catch to ensure that if anything goes wrong, the timeout task still executes and kills this job ungracefully.